src/utils/fs.ts

```diff
@@ -50,6 +50,34 @@ export function isFile (path: string): Promise<boolean> {
 }
 
 /**
+ * Verify a path exists and is a directory.
+ */
+export function isDirectory (path: string): Promise<boolean> {
+  return stat(path).then(stat => stat.isDirectory(), () => false)
+}
+
+function filterAsync<T>(items: T[], predicate: { (item: T): Promise<boolean> }): Promise<T[]> {
+  const pendingFilters = items.map(item => predicate(item))
+
+  return Promise.all(pendingFilters)
+    .then(filters => {
+      return items.filter((item: T, index: number) => filters[index])
+    })
+}
+
+/**
+ * Read the sub directories under in a path.
+ */
+export function readSubDirs (path: string): Promise<string[]> {
```

Comment:
Yeah, I don't know. I thought it'd be nicer to use it as a source of truth, though I did consider doing a glob for endpoints too - it'd be easier if we save `typings.json` files, which I've been meaning to add. 

You aren't currently search all the typings folders right, this is just if one is not there when `main.d.ts` says it should be? I'd just use `main.d.ts` as the source of truth, map over negatives in `typings.json`, run `rimraf` on the folders. You can also check the file exists like the behaviour in `uninstall`: https://github.com/typings/core/blob/master/src/uninstall.ts#L100.